### PR TITLE
Add script to generate the license file

### PR DESCRIPTION
**Pull request by johnson325**

### CODE REVIEW

File: meta-iot2000-example/scripts/bin/extract_license [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# Extract and combine license file
+# Eg:
+#    extract_license build/tmp/deploy/licenses license.dat
+#
```
Comment (jan-kiszka) [inline meta-iot2000-example/scripts/bin/extract_license]: Copyright and license header is missing. We didn't use that consistently so far, specifically leaving trivial one or two-line files untagged, but we should do better. Grab the pattern from meta-iot2000-bsp/recipes-tools/switchmode/files/switchmode.c e.g.

File: meta-iot2000-example/scripts/bin/extract_license [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+dir_path="${1:?Please provide license directory path}"
+output_path="${2:?Please provide output file path}"
+dir_list=$(get_dir_list $dir_path)
+truncate $output_path --size 0
+for dir in $dir_list; do
+    echo "Component: "$dir >> $output_path
+    file_list=$(get_file_list $dir_path"/"$dir"/")
+    for file in $file_list; do
+    file_path=$dir/$file
```
Comment (jan-kiszka) [inline meta-iot2000-example/scripts/bin/extract_license]: Please indent this for loop block properly.

File: meta-iot2000-example/scripts/bin/extract_license [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+function get_dir_list()
+{
+    dir=$(find $1/* -type d -printf "%f\n")
+    echo $dir
+}
+
+function get_file_list()
+{
+
```
Comment (jan-kiszka) [inline meta-iot2000-example/scripts/bin/extract_license]: Unneeded newline

**Review by jan-kiszka — CHANGES_REQUESTED** [review]
Is scripts/*bin*/ a typical path here? I would just put it in .../scripts.

**Comment by jan-kiszka** [comment]
Please fold your fixes into the the original commit (no need to merge review history). And don't forget to sign-off the commit.

**Comment by johnson325** [comment]
Done. Thanks for the comments.

**Pull request merged**